packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java

```diff
@@ -314,6 +305,20 @@ private Set<ViewModifier> computeModifiers(Optional<IDiagramEvent> optionalDiagr
         return modifiers;
     }
 
+    private void computeLayoutModifiers(Optional<IDiagramEvent> optionalDiagramEvent, String nodeId, Set<ViewModifier> modifiers) {
+        optionalDiagramEvent.ifPresent(diagramEvent -> {
+            if (diagramEvent instanceof PinDiagramElementEvent pinDiagramElementEvent) {
+                if (pinDiagramElementEvent.getElementIds().contains(nodeId)) {
+                    if (pinDiagramElementEvent.pinElement()) {
+                        modifiers.add(ViewModifier.Pinned);
+                    } else {
+                        modifiers.remove(ViewModifier.Pinned);
+                    }
+                }
+            }
+        });
+    }
```

Comment:
That's a lot of nested `if` for something starting with an optional, the whole function can be inlined quite easily

Reply:
This could simply be a method `private boolean isPinned(Optional<IDiagramEvent> optionalDiagramEvent, String nodeId)` which would return the pin status. Here you are adding something in `modifiers` to read it a couple lines after.

Reply:
Done